### PR TITLE
RET-829

### PR DESCRIPTION
**Pull request by hsjhita**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RET-829

### Change description ###

New event added to transfer case to ECM

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/casetransfer/CaseTransferToEcmParams.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@Getter
+public class CaseTransferToEcmParams {
+    private String userToken;
+    private String caseTypeId;
+    private String jurisdiction;
+    private List<String> ethosCaseReferences;
+    private String newCaseTypeId;
+    private String positionType;
+    private String reason;
+    private String multipleReference;
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/casetransfer/CaseTransferToEcmParams.java:18]: Does multipleReference serve any purpose? If not, can we get rid of it and any others that aren't used?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/casetransfer/CaseTransferToEcmServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    private CaseTransferToEcmService caseTransferToEcmService;
+
+    @Mock
+    private CaseTransferUtils caseTransferUtils;
+
+    @Mock
+    private CaseTransferEventService caseTransferEventService;
+
+    @Test
+    void transferTOEcm() {
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/casetransfer/CaseTransferToEcmServiceTest.java:41]: To not TO

File: build.gradle [hunk 3]
```diff
@@ -21,7 +21,8 @@ allprojects {
 }
 
 checkstyle {
-    maxWarnings = 4358
+    maxWarnings = 5000
```
[the remark this answers is not in the document]
Comment (hsjhita) [reply build.gradle]: Was initially increased due to old tests causing an issue but will see. Reverted back to original figure

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/casetransfer/CaseTransferToEcmParams.java (hunk 1 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/casetransfer/CaseTransferToEcmParams.java]: Parameters removed

**Review by jthmcts — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/casetransfer/CaseTransferToEcmServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    @Test
+    void noCasesFound() {
+        var ecmOfficeCT = TribunalOffice.BRISTOL.getOfficeName();
+        var caseDetails = createCaseDetails(TribunalOffice.LEEDS.getOfficeName(), ecmOfficeCT);
+        when(caseTransferUtils.getAllCasesToBeTransferred(caseDetails, AUTH_TOKEN))
+                .thenReturn(Collections.emptyList());
+        var errors = caseTransferToEcmService.createCaseTransferToEcm(caseDetails, AUTH_TOKEN);
+        assertFalse(errors.isEmpty());
+        assertEquals(1, errors.size());
```
Comment (robin-ieong) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/casetransfer/CaseTransferToEcmServiceTest.java:60]: small one but I think we can just check the errors size without assertFalse isEmpty

Comment (hsjhita) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/casetransfer/CaseTransferToEcmServiceTest.java:59]: Assertion removed

**Review by robin-ieong — APPROVED** [review]

**Pull request merged**